ReCord_frontend.py

```diff
@@ -92,49 +103,44 @@ def search_terms(tag, para, tree):
     Return: rendered result page 'ReChord_result.html'
     """
 
-    if tag == 'Expressive Terms':
-        # todo: do search on expressive terms
-        result = find_artic(tree, para)
-    elif tag == 'Articulation':
-        result = find_expressive_term(tree.root, para)
-
-    # todo: Integrate more term search
-    # elif tag == 'Tempo Marking':
-    # elif tag == 'Dynamic Marking':
-    # elif tag == 'Piano Fingerings':
-    # elif tag == 'Pedal Marking':
-    # elif tag == 'Hairpin':
-    # elif tag == 'Slur/Ligatures':
-    # elif tag == 'Ornaments':
-    # elif tag == 'Notes':
-    # elif tag == 'Accidental':
-    return render_template('ReChord_result.html', result=result)
+    return render_template('ReChord_result.html', result=text_box_search_folder(path, tag, para))
 
 
 def upload_file(name_tag):
-    """pass the upload file and store it in uploads folder
+    """pass the upload files and store them in uploads folder's unique sub-folder
     Arguments: name_tag that used in html
-    Return: rendered result page 'ReChord_result.html'"""
+    Return: upload path name
+    """
 
     # check if the post request has the file part
     if 'base_file' not in request.files:
         flash('No file part')
         return redirect(request.url)
     else:
-        file = request.files[name_tag]
-
-        # if user does not select file, browser also submit a empty part without filename
-        if file.filename == '':
-            flash('No selected file')
-            return redirect(request.url)
-
-        # if properly uploaded
-        elif file and allowed_file(file.filename):
-            filename = secure_filename(file.filename)
-            file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
-            # todo handle return
-            return filename
-
+        files = request.files.getlist(name_tag)
+        file_path = make_upload_dir()
+
+        for file in files:
+            # if user does not select file, browser also submit a empty part without filename
+            if file.filename == '':
+                flash('No selected file')
+                return redirect(request.url)
+
+            # if properly uploaded
+            elif file and allowed_file(file.filename):
+                filename = secure_filename(file.filename)
+                file.save(os.path.join(file_path, filename))
+        return file_path
+
+
+def make_upload_dir():
+    """return a unique file_path for each upload
+    RETURN: unique file path under uploads folder"""
+
+    file_path = r"database/uploads/" + str(uuid.uuid4()) + "/"
+    directory = os.path.dirname(file_path)
+    os.makedirs(directory)
```

Comment:
Use the `tempfile` library instead: https://docs.python.org/dev/library/tempfile.html